internal/p2p/peermanager.go

```diff
@@ -799,15 +802,27 @@ func (m *PeerManager) Errored(peerID types.NodeID, err error) {
 	m.evictWaker.Wake()
 }
 
+func (m *PeerManager) Inactivate(peerID types.NodeID) error {
+	m.mtx.Lock()
+	defer m.mtx.Unlock()
+
+	peer := m.store.peers[peerID]
+	peer.Inactive = true
+
+	return m.store.Set(*peer)
+}
```

Comment:
Isn't the concept of inactiveness captured by the `DialFailures` field

Reply:
Inactiveness seems to be smaller (the only place we're using it now, is where we had been deleting known peers, marking them as inactive means we don't gossip them and we don't try and connect to them, but we avoid deleting them so that we can't get "posioned" by another peer store. 

should probably mark more things inactive.

Reply:
Ok so `Ranked()` won't include "Inactive" peers?